columnflow/plotting/plot_util.py

```diff
@@ -55,6 +58,87 @@ def get_cms_label(ax: plt.Axes, llabel: str) -> dict:
     return cms_label_kwargs
 
 
+def round_dynamic(value: int | float) -> int | float:
+    """
+    Rounds a *value* at various scales to a subjective, sensible precision. Rounding rules:
+
+        - 0 -> 0
+        - (0, 1) -> round to 1 significant digit
+        - [1, 20]: round to 1
+        - (20, 100]: round to 5
+        - (100, 500]: round to 10
+        - (500, 1000]: round to 50
+        - (1000, 5000]: round to 100
+        - (5000, 10000]: round to 500
```

Comment:
```suggestion
        - (100, 200]: round to 10
        - (200, 1000]: round to 50
        - (1000, 2000]: round to 100
        - (2000, 10000]: round to 500
```

Reply:
Mh, I've been thinking about this again, and I'd vote for re-introducing the 20/200 steps since maximum possible divergence is just too large. Say, at a value of 225, the rounding to 200 or 250 would cause a change of roughly 12.5%, which is quite a lot.

Reply:
After talking to @dsavoiu we came up with a simpler and more accurate way. We could just round to 1 significant digits below 10, and to two significant digits above:

- 0 -> 0 (int)
- (0, 1) -> round to 1 significant digit (float)
- [1, 10) -> round to 1 significant digit (int)
- [10, inf): round to 2 significant digits (int)

Reply:
That's much easier to comprehend, I like it. The only cases one could still discuss are the steps from e.g. 1 to 2 or 0.1 to 0.2, since there we can still get a rounding of 50%. But it's probably fine as well since it is now clear that we round to 1 significant digit.

Reply:
Indeed. And these additional cases are maybe quite use-case specific. Maybe let's revisit them once we stumble upon them.